src/Giraffe/EndpointRouting.fs

```diff
@@ -156,29 +157,71 @@ module Routers =
             | CONNECT -> "CONNECT"
             | _ -> ""
 
+    [<RequireQualifiedAccess>]
+    type AspNetExtension =
```

Comment:
I'm not a fan the DU. This means if someone wants to extend this functionality they have to edit giraffe source code. And if they're doing something custom it's not something we'd want to accept. 

It would be better to have a list of `IEndpointConventionBuilder -> IEndpointConventionBuilder/unit` functions and maybe provide helpers for known ones we'd want to support.